Sources/DataPipeline/Config/SDKConfigBuilder.swift

```diff
@@ -0,0 +1,147 @@
+import CioInternalCommon
+import Foundation
+import Segment
+
+/// Builder class designed to facilitate the creation of `SdkConfig` and `DataPipelineConfigOptions`.
+/// The builder pattern allows for a fluent and intuitive way to set up configuration options for
+/// the SDK, ensuring all required and optional settings are appropriately configured before the
+/// SDK is initialized.
+///
+/// Usage Example:
+/// ```
+/// let config = SDKConfigBuilder(writeKey: "your_write_key")
+///   .apiHost("your_api_host")
+///   .flushAt(30)
+///   // additional configuration as needed...
+///   .build()
+/// // Use `config` for initializing the SDK...
+/// ```
+public class SDKConfigBuilder {
```

Comment:
```suggestion
public struct SDKConfigBuilder {
```

Because this class's job is mostly to hold onto properties and modify them, can we make this a `struct`? 

A swift struct vs class is similar to Kotlin's class vs data class conversation. data classes/structs provide some benefits that could be useful for us in the future.

Reply:
Thanks for sharing, updated.

Reply:
I've to revert this, because it fails when using chaining. The code in `APITest` looks like

```
SDKConfigBuilder(writeKey: "")
  .apiHost("")
  .cdnHost("")
  .flushAt(10)
  .flushInterval(10.0)
  .autoAddCustomerIODestination(false)
  .defaultSettings(Settings(writeKey: ""))
  .flushPolicies([])
  .addFlushPolicies([CountBasedFlushPolicy(), IntervalBasedFlushPolicy()])
  .flushQueue(DispatchQueue(label: ""))
  .operatingMode(OperatingMode.asynchronous)
  .trackApplicationLifecycleEvents(true)
  .autoTrackDeviceAttributes(true)
```
and fails with following error with struct

```
Cannot use mutating member on immutable value: function call returns immutable value
```